addons/payment_stripe/static/src/js/payment_form.js

```diff
@@ -42,7 +43,250 @@ odoo.define('payment_stripe.payment_form', require => {
+            // Check if instantiation of the drop-in is needed
+            if (this.StripeExpress && this.StripeExpress.acquirerId === paymentOptionId) {
+                return Promise.resolve(); // Don't re-instantiate if already done for this acquirer
+            }
+            var self = this;
+            self._rpc({
+                // Create transaction
+                route: '/payment/stripe/publishable_key',
+                params: {acquirer_id:paymentOptionId},
+            }).then(stripe_publishable_key => {
+                const stripeJS = Stripe(stripe_publishable_key);
+                var isShippingInformationRequired = self._isShippingInformationRequired()
+                var paymentRequest = stripeJS.paymentRequest({
+                    country:'BE',
+                    currency: this.txContext.currencyName,
+                    total: {
+                        label: this.txContext.label,
+                        amount: this.txContext.minorAmount,
+                        pending: isShippingInformationRequired,
+                    },
+                    displayItems: isShippingInformationRequired ? [
+                        {
+                            label: 'Your order',
+                            amount: this.txContext.minorAmount,
+                        },
+                        {
+                            label: 'Delivery',
+                            amount: 0,
+                            pending: false,
+                        },
+                    ]:[],
+                    requestPayerName: true,
+                    requestPayerEmail: true,
+                    requestPayerPhone: true,
+
+                    requestShipping: isShippingInformationRequired,
+                });
+                self.StripeExpress = paymentRequest;
+                self.StripeExpress.acquirerId = paymentOptionId;
+
+                var elements = stripeJS.elements();
+                var prButton = elements.create('paymentRequestButton', {
+                    paymentRequest: paymentRequest,
+                    style: {
+                        paymentRequestButton: {
+                          type: 'buy',
+                          // One of 'default', 'book', 'buy', or 'donate'
+                          // Defaults to 'default'
+                        },
+                      },
+                });
+
+                // Check the availability of the Payment Request API first.
+                paymentRequest.canMakePayment().then(function(result) {
+                    if (result) {
+                        prButton.mount(`#o_stripe_express_checkout_container_${paymentOptionId}`);
+                    }
+                });
+
+                paymentRequest.on('paymentmethod', function(ev) {
+                    console.log(ev);
+                    var addresses = {}
+                    addresses['partner_id'] = parseInt(self.txContext.partnerId)
+                    addresses['billing_address'] = {
```

Comment:
It's a bit weird to create an empty object then start filling it with literal string keys, why not create the object with the two keys directly?